libraries/Adapters/Microsoft.Bot.Builder.Adapters.Slack/SlackClientWrapper.cs

```diff
@@ -28,19 +29,21 @@ public SlackClientWrapper(string botToken)
         /// <param name="name">The optional name.</param>
         /// <param name="channel">The optional channel.</param>
         /// <param name="timestamp">The optional timestamp.</param>
+        /// <param name="cancellationToken">A cancellation token for the task.</param>
         /// <returns>The <see cref="ReactionAddedResponse"/> representing the response to the reaction added.</returns>
-        public virtual async Task<ReactionAddedResponse> AddReactionAsync(string name = null, string channel = null, string timestamp = null)
+        public virtual async Task<ReactionAddedResponse> AddReactionAsync(string name = null, string channel = null, string timestamp = null, CancellationToken cancellationToken = default(CancellationToken))
         {
             return await _api.AddReactionAsync(name, channel, timestamp).ConfigureAwait(false);
         }
 
         /// <summary>
         /// Wraps Slack API's APIRequestWithTokenAsync method.
         /// </summary>
+        /// <param name="cancellationToken">A cancellation token for the task.</param>
         /// <param name="postParameters">The parameters to the POST request.</param>
         /// <typeparam name="TResult">The generic type for the return. Must be of type Response.</param>
         /// <returns>A <see cref="Task"/> of type T representing the asynchronous operation.</returns>
-        public virtual async Task<TResult> APIRequestWithTokenAsync<TResult>(params Tuple<string, string>[] postParameters)
+        public virtual async Task<TResult> APIRequestWithTokenAsync<TResult>(CancellationToken cancellationToken, params Tuple<string, string>[] postParameters)
```

Comment:
CancellationToken must always be the last parameter.

Reply:
tell that to **params** https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/params